### DISCUSSION TITLE
Laravel Echo not found warning when listening for event in Livewire Component

### DISCUSSION THREAD

**Discussion by hyphen81** [livewire/livewire#5204] [1. Help]
I'm successfully broadcasting an event and getting a response back from Pusher. I can log the response in the console by including

`Echo.private(MyChannel.1).listen('MyEvent', (e) => {console.log(e)});`

and I get a valid response.

However, as soon as I add `"echo-private:statements.{$this->statement->id},StatementCompleted" => 'testingThis',` to the `getListeners` method in my Livewire component, I get the warning `LaravelEcho.js:9 Laravel Echo cannot be found`.

Commenting out the above listener line causes everything to work fine. I've been over the docs what feels like a dozen times to make sure I don't have a stupid type-o or something, and I'm totally at a loss.

I have a SO question here: [](https://stackoverflow.com/questions/73472255/laravel-echo-echo-cannot-be-found)

**Comment by themahdscientist** [comment]
encountered the same error and still no answer after almost **3 years** huh? @danharrin

**Reply by lrljoe** [reply]
Assuming you're using v3

```php
#[On('echo:orders,OrderShipped')]
public function notifyNewOrder()
{
    $this->showNewOrderNotification = true;
}
```

Is the approach to listen for Echo events.

The "broken" approach that was posted 2 years ago is almost certainly due to the property not existing at load.  Trying to listen to something that is null isn't going to work.
